### PR TITLE
Redo README in the style of https://opensource.envato.com/checklist/

### PR DESCRIPTION
**Pull request by johnsyweb**
#### Context

Preparing to release #28 in v1.1, I noted the README didn't match our current [Open Source Guidelines](https://opensource.envato.com/checklist/).

#### Change

 - Redo README in the style of https://opensource.envato.com/checklist/
 - Add Code of Conduct from https://www.contributor-covenant.org/

### CODE REVIEW

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
+
+# Contributor Covenant Code of Conduct
```
Comment (lucascaton) [inline CODE_OF_CONDUCT.md:2]:
```suggestion
# Contributor Covenant Code of Conduct
```

File: README.md [hunk 2]
```diff
@@ -107,3 +120,60 @@ $ irb
...
+
+For bug fixes, documentation changes, and features:
+
+1. [Fork it](./fork)
+1. Create your feature branch (`git checkout -b my-new-feature`)
+1. Commit your changes (`git commit -am 'Add some feature'`)
+1. Push to the branch (`git push origin my-new-feature`)
+1. Create a new Pull Request
+
+For larger new features: Do everything as above, but first also make contact with the project maintainers to be sure your change fits with the project direction and you won't be wasting effort going in the wrong direction.
```
Comment (lucascaton) [inline README.md:161]: Should we explain how to contact the project maintainers?

File: CODE_OF_CONDUCT.md (hunk 1 above)
Comment (johnsyweb) [reply CODE_OF_CONDUCT.md:2]: This comes directly from https://github.com/ContributorCovenant/contributor_covenant/blob/release/CODE_OF_CONDUCT.md . I'd prefer to keep the whitespace consistent for future diffs.

File: README.md (hunk 2 above)
Comment (johnsyweb) [reply README.md:161]: Is this not covered by 670e217 and a7f6ab5 ?

Comment (thenathanjones) [reply README.md:161]: Yeah that feels sufficient to me.

Comment (lucascaton) [reply README.md:161]: I meant to add a link here :)

```suggestion
For larger new features: Do everything as above, but first also [make contact with the project maintainers](https://github.com/envato/zxcvbn-ruby#contact) to be sure your change fits with the project direction and you won't be wasting effort going in the wrong direction.
```

**Pull request merged**